packages/@aws-cdk/aws-dynamodb/lib/table.ts

```diff
@@ -188,6 +272,29 @@ export class Table extends Construct {
         return errors;
     }
 
+    /**
+     * Validate non-key attributes by checking limits within secondary index, which may vary in future.
+     *
+     * @param {string[]} nonKeyAttributes a list of non-key attribute names
+     */
+    private validateNonKeyAttributes(nonKeyAttributes: string[]) {
+        if (this.attributeDefinitions.length + this.nonKeyAttributes.length + nonKeyAttributes.length > 20) {
+            // https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/Limits.html#limits-secondary-indexes
+            throw new RangeError('a maximum number of nonKeyAttributes across all of secondary indexes is 20');
```

Comment:
Is it correct to include `attributeDefinitions` in this computation?

Reply:
Thanks for the question! I tested this again and `attributeDefinitions` should not be there.

I will add a case where creating 20 non-key attributes to the integration test.